eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -179,36 +179,43 @@ jobs:
               return $line
             }
 
-            $dependencies | Out-File $env:REQUIREMENTS
+            $dependencies | Out-File $(requirements)
 
           displayName: Override requirements with pipeline build artifact versions
-          env:
-            REQUIREMENTS: $(requirements)
 
-        - pwsh: pip install -r "$env:REQUIREMENTS" --no-deps --upgrade
+        - pwsh: |
+            $ErrorActionPreference = "Continue"
+            while ($retries++ -lt 4) {
+              Write-Host "pip install -r $(requirements) --no-deps --upgrade --no-cache-dir"
+              pip install -r "$(requirements)" --no-deps --upgrade --no-cache-dir
+              if ($LASTEXITCODE) {
+                if ($retries -ge 4) {
+                  exit $LASTEXITCODE
+                }
+                Write-Host "Installation failed, retrying in 3 minutes..."
+                sleep 180
+              } else {
+                $ErrorActionPreference = "Stop"
```

Comment:
I think we can remove this one as well.

Reply:
Can we? won't that affect the error action for the powershell scripts that executes after this?

Reply:
Or are the pwsh settings reset after each step (because we also have more pip installs in the next couple consecutive steps, and they do rely on $ErrorActionPreference for outputting the errors and exiting)

Reply:
Yup it is separated between different steps

Reply:
Yes each step starts its own powershell process so they don't impact each other.